src/query/management/src/role/role_mgr.rs

```diff
@@ -158,10 +173,18 @@ impl RoleApi for RoleMgr {
 
         match seq.match_seq(&seq_value) {
             Ok(_) => {
-                let data = serde_json::from_slice::<RoleInfo>(&seq_value.data).or_else(|err| {
-                    log::debug!("serde json err when get role. cause : {}", err);
-                    deserialize_struct(&seq_value.data, ErrorCode::IllegalUserInfoFormat, || "")
+                let mut need_serialize_pb = false;
+                let data = deserialize_struct(&seq_value.data, ErrorCode::IllegalUserInfoFormat, || "").or_else(|err| {
+                    log::debug!("deserialize RoleInfo as pb err when get role. cause : {}, rollback to use serde json", err);
+                    need_serialize_pb = true;
+                    serde_json::from_slice::<RoleInfo>(&seq_value.data)
```

Comment:
We should check the `serde_json::from_slice::<RoleInfo>(&seq_value.data)` is OK, then set `need_serialize_pb = true`, or it may be a pb deserialize error.

Reply:
Prefer move this logic to a fn like named `check_and_upgrade_to_pb`:
```
pub fn check_and_upgrade_to_pb<T>(...) ...{
 let mut need_serialize_pb = false;
                let data = deserialize_struct(&seq_value.data, ErrorCode::IllegalUserInfoFormat, || "").or_else(|err| {
                    log::debug!("deserialize RoleInfo as pb err when get role. cause : {}, rollback to use serde json", err);
                   if OK(..) serde_json::from_slice::<RoleInfo>(&seq_value.data) {
                                          need_serialize_pb = true;
                                          ...
                   }
                   
                   if need_serialize_pb {
                     upgrade_to_pb(
                   }
                   ...
}
```

Then all others which is json format can use this `check_and_upgrade_to_pb` to upgrade.

Reply:
> We should check the `serde_json::from_slice::<RoleInfo>(&seq_value.data)` is OK, then set `need_serialize_pb = true`, or it may be a pb deserialize error.

If there is a json decoding error, `need_serialize_pb` won't take effect, IMO(as indicated in the following code snippet).
Admittedly, the logic in this instance may not be immediately apparent.

```
                let data = deserialize_struct(&seq_value.data, ErrorCode::IllegalUserInfoFormat, || "").or_else(|err| {
                    log::debug!("deserialize RoleInfo as pb err when get role. cause : {}, rollback to use serde json", err);
                    need_serialize_pb = true;
                    serde_json::from_slice::<RoleInfo>(&seq_value.data)
                })?;

                if need_serialize_pb {
                    let value = serialize_struct(&data, ErrorCode::IllegalUserInfoFormat, || "")?;
                    self.upgrade_to_pb(key, value, MatchSeq::Exact(seq_value.seq))
                        .await?;
                }
                Ok(SeqV::new(seq_value.seq, data))
```

Reply:
And plus:
`check_and_upgrade_to_pb` is easy to do unit test. Logic test is more harder to check this case.

Reply:
Done.